pkg/cmd/bootstrap.go

```diff
@@ -123,9 +125,45 @@ func nonInteractiveMode(io *BootstrapParameters, client *utility.Client) error {
 	if err := checkMandatoryFlags(mandatoryFlags); err != nil {
 		return err
 	}
+	if io.GitHostAccessToken != "" {
+		err := setSecret(io.GitOpsRepoURL, io.GitHostAccessToken)
+		if err != nil {
+			return err
+		}
+		err = setSecret(io.ServiceRepoURL, io.GitHostAccessToken)
+		if err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
+func setSecret(repoURL, accessToken string) error {
+	secret, hostName, err := getSecretFromRepoURL(repoURL)
+	if err != nil && err != keyring.ErrNotFound {
+		return err
+	}
+	if accessToken != secret {
+		err := keyring.Set(webhook.KeyringServiceName, hostName, accessToken)
+		if err != nil {
+			return fmt.Errorf("unable to set access token for repo %q using keyring: %w", repoURL, err)
+		}
+	}
+	return nil
+}
+
+func getSecretFromRepoURL(repoURL string) (string, string, error) {
```

Comment:
I see a problem is this fuction.   It can return err (not found) and yet this caller can use the returned value `hostname`.   The problem is that it becomes unclear when and which returned values the caller can use when there is an error.   
I would restructure the fuction to something like this: 

```
// getSecret returns secret from keyring service by the given hostName.  If secret is not found (keyring.ErrNotFound), it returns "".
func getSecret(hostName string) (string, error)
```

It won't return error if not found. 

And you call another method to get hostname from url which is not part of this method.